tests/legacy-cli/e2e/utils/registry.ts

```diff
@@ -17,16 +17,19 @@ export async function createNpmRegistry(
   );
   configContent = configContent.replace(/\$\{HTTP_PORT\}/g, String(port));
   configContent = configContent.replace(/\$\{HTTPS_PORT\}/g, String(httpsPort));
-  await writeFile(join(registryPath, 'verdaccio.yaml'), configContent);
+  const configPath = join(registryPath, 'verdaccio.yaml');
+  await writeFile(configPath, configContent);
 
-  return spawn(
-    process.execPath,
-    [require.resolve('verdaccio/bin/verdaccio'), '-c', './verdaccio.yaml'],
-    {
-      cwd: registryPath,
-      stdio: 'inherit',
-    },
-  );
+  const verdaccioServer = fork(require.resolve('verdaccio/bin/verdaccio'), ['-c', configPath]);
+  await new Promise<void>((done) => {
+    verdaccioServer.on('message', (msg: { verdaccio_started: boolean }) => {
+      if (msg.verdaccio_started) {
+        done();
+      }
+    });
+  });
```

Comment:
Should be able to do something like the following (`on` is from `node:events`):
```
for await (const event of on(verdacioServer, 'message')) {
    if (event.verdaccio_started) {
        break;
    }
}
```

Could even add a timeout as well: `{ signal: AbortSignal.timeout(30_000) }` for the third parameter of `on`